go.mod

```diff
@@ -46,6 +46,7 @@ require (
 	github.com/prometheus/client_model v0.6.2 // indirect
 	github.com/prometheus/common v0.65.0 // indirect
 	github.com/prometheus/procfs v0.16.1 // indirect
+	github.com/puzpuzpuz/xsync/v4 v4.1.0
```

Comment:
The dependency added here provides a proven implementation supported by an active developer community.